### PR TITLE
Add GitHub Actions information, and fixes some syntax highlighting

### PR DESCRIPTION
**Pull request by frederikspang**

### CODE REVIEW

**Review by mads-bo — COMMENTED** [review]
@aau-giraf/sw607f20 you have been assigned this. Here's your checklist.

Please use the following checklist:

**Code Documentation Only**

- [ ] It is documented which BLoCs/Screens the current class is related to
- [ ] The responsibility/functionality of the current class is explained. Also regarding the classes it is related to
- [ ] If any unusual decisions are introduced in the code, they reasoned about in the documentation
 
**Wiki Related**

- [ ] There are no dead links or references
- [ ] The markdown renders correctly in MkDocs
- [ ] The content is placed in the correct page/folder/section

You can run MkDocs locally by following [this guide](https://aau-giraf.github.io/wiki/writing_wiki/#running-locally)

**Review by mads-bo — COMMENTED** [review]
@aau-giraf/sw611f20 you have been assigned this. Here's your checklist.

Please use the following checklist:

**Code Documentation Only**

- [ ] It is documented which BLoCs/Screens the current class is related to
- [ ] The responsibility/functionality of the current class is explained. Also regarding the classes it is related to
- [ ] If any unusual decisions are introduced in the code, they reasoned about in the documentation
 
**Wiki Related**

- [ ] There are no dead links or references
- [ ] The markdown renders correctly in MkDocs
- [ ] The content is placed in the correct page/folder/section

You can run MkDocs locally by following [this guide](https://aau-giraf.github.io/wiki/writing_wiki/#running-locally)

File: docs/development/continuous_integration/github_actions.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# GitHub Actions
+The currently used Continous Integration Pipeline is GitHub Action, which provides 2.000 minutes free per month, running on either Ubuntu, MacOS or Windows, allowing us to test multiple platforms from our CI-Platform without cost.
+
+## Weekplanner
+The weekplanner built in 4 jobs:
+
+ - `build-and-test-android`
+ - `build-and-test-ios`
+ - `deploy-to-playstore`
+ - `deploy-to-appstore`
```
Comment (joandrsn) [inline docs/development/continuous_integration/github_actions.md:10]: Could it be an idea to link directly to the files in this list?

File: docs/development/continuous_integration/github_actions.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+Of these jobs, the `build-and-test-android` job runs `flutter test` and `flutter build appbundle`, is run across all platforms, and produces an appbundle file, ready for upload to Google Play Store, in the `deploy-to-playstore`-task, which is dependent on this task, and furthermore only runs on master.
```
Comment (joandrsn) [inline docs/development/continuous_integration/github_actions.md]: Very long sentence. Hard to follow

File: docs/development/continuous_integration/github_actions.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+Due to the buildability, and iOS build requires XCode as a depedency, task `build-and-test-ios` only runs on MacOS, and produces a `app.ipa`-file, iOS App Store Package, and prepares this for `deploy-to-appstore`, that uploads a build to our App Store Connect-account, which can be pushed to internal test, or into release state.
```
Comment (joandrsn) [inline docs/development/continuous_integration/github_actions.md]: Also very long and could be difficult to understand for new students.

File: docs/development/continuous_integration/github_actions.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+### Docker Hub
+In addition to running unit tests in GitHub Actions, when pushed to `develop`- or `master`-branches Docker Hub picks up a webhook, and starts a build of the embedded Dockerfile, and tags this accordingly as either:
+
+ - develop-1-aspnet
+ - master-1-aspnet
+
+When deploying into the [production swarm](../../server_administration/PractiaclDocker), these images are pulled into DEV and PROD environments accordingly.
+
+## wiki
+As the wiki is currently built using MKDocs, this repository is setup with a GitHub Actions script to build the source-files, markdown, into a static HTML page, and push this into the GitHub Pages-environment, allowing it to be reachable at [http://giraf-aau.github.io/wiki](giraf-aau.github.io/wiki).
```
Comment (joandrsn) [inline docs/development/continuous_integration/github_actions.md]: Compared to the others, this one is very general.

**Review by joandrsn — CHANGES_REQUESTED** [review]
I think the text on continuous integration could be better. If there is something you disagree with, either hit me up or write a response.

File: docs/development/continuous_integration/github_actions.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
+# GitHub Actions
+The currently used Continous Integration Pipeline is GitHub Action, which provides 2.000 minutes free per month, running on either Ubuntu, MacOS or Windows, allowing us to test multiple platforms from our CI-Platform without cost.
```
Comment (antonhpalmer) [inline docs/development/continuous_integration/github_actions.md]:
```suggestion
The currently used Continuous Integration Pipeline is GitHub Actions, which provides 2.000 minutes free per month, running on either Ubuntu, MacOS or Windows, allowing us to test multiple platforms from our CI-Platform without cost.
```

File: docs/development/continuous_integration/github_actions.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+## Weekplanner
+The weekplanner built in 4 jobs:
```
Comment (antonhpalmer) [inline docs/development/continuous_integration/github_actions.md]:
```suggestion
The weekplanner is built in 4 jobs:
```

File: docs/development/continuous_integration/github_actions.md (hunk 3 above)
Comment (antonhpalmer) [inline docs/development/continuous_integration/github_actions.md]:
```suggestion
Due to the buildability, and the fact that iOS build requires XCode as a dependency, task `build-and-test-ios` only runs on MacOS, and produces an `app.ipa`-file, iOS App Store Package, and prepares this for `deploy-to-appstore`, which uploads a build to our App Store Connect-account, which can be pushed to internal test, or into release state.
```

File: docs/development/continuous_integration/github_actions.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+## web-api
+Similar to the api_client, the web-api suite runs by building a dotnet system in Release mode, to ensure deployability, and then runs `dotnet test`, to run the embedded GirafRest.Test project and the unit tests contained.
+
+### Docker Hub
+In addition to running unit tests in GitHub Actions, when pushed to `develop`- or `master`-branches Docker Hub picks up a webhook, and starts a build of the embedded Dockerfile, and tags this accordingly as either:
+
+ - develop-1-aspnet
+ - master-1-aspnet
+
+When deploying into the [production swarm](../../server_administration/PractiaclDocker), these images are pulled into DEV and PROD environments accordingly.
```
Comment (antonhpalmer) [inline docs/development/continuous_integration/github_actions.md]:
```suggestion
When deploying into the [production swarm](../../server_administration/PracticalDocker), these images are pulled into DEV and PROD environments accordingly.
```

File: docs/development/continuous_integration/github_actions.md (hunk 4 above)
Comment (antonhpalmer) [inline docs/development/continuous_integration/github_actions.md]:
```suggestion
As the wiki is currently built using MKDocs, this repository is set up with a GitHub Actions script to build the source files, markdown, into a static HTML page, and push this into the GitHub Pages-environment, allowing it to be reachable at [http://giraf-aau.github.io/wiki](giraf-aau.github.io/wiki).
```

File: docs/development/continuous_integration/github_actions.md (hunk 7 above)
Comment (antonhpalmer) [reply docs/development/continuous_integration/github_actions.md]: the file should be renamed

**Review by joandrsn — DISMISSED** [review]

File: docs/development/server_administration/docker_swarm/Database.md [hunk 8]
```diff
@@ -4,9 +4,9 @@ In order for the new Docker Swarm to serve the Giraf project, the database from
 
 ## Old setup
 
-The old server was configured to use MySql 5.7.11 as can be seen on the picture below.
+The old server was configured to use MySQL 8.0.19 as can be seen on the picture below.
```
Comment (antonhpalmer) [inline docs/development/server_administration/docker_swarm/Database.md]: The image still says version 5.7.11

**Review by antonhpalmer — CHANGES_REQUESTED** [review]
a few comments

File: docs/development/server_administration/docker_swarm/Database.md (hunk 8 above)
Comment (frederikspang) [reply docs/development/server_administration/docker_swarm/Database.md]: Not related to the issue per se. Fixed this, because I saw it there, and I knew we updated. There IS an issue in backlog for ensuring consistent documentation in regards to versioning.

Comment (antonhpalmer) [reply docs/development/server_administration/docker_swarm/Database.md]: But this is during the explanation of the "Old setup", so I don't think the version number should be changed

Comment (frederikspang) [reply docs/development/server_administration/docker_swarm/Database.md]: You're very right. Corrected!

**Review by antonhpalmer — APPROVED** [review]

**Review by joandrsn — APPROVED** [review]
